workitem/link/link_repository.go

```diff
@@ -117,7 +117,7 @@ func (r *GormWorkItemLinkRepository) ValidateTopology(ctx context.Context, sourc
 			return errs.Wrapf(err, "error during cycle-detection of new link")
 		}
 		if hasCycle {
-			return errs.New("link cycle detected")
+			return errors.NewBadParameterError("sourceID + targetID", fmt.Sprintf("%s + %s", sourceID, targetID)).Expected("link that doesn't create a cylce")
```

Comment:
How about using 409? `errors.NewDataConflictError(...)` The cyclic linking can be considered as a conflict?
https://httpstatuses.com/409

Reply:
Already done @baijum as it was being suggested by @aslakknutsen 5e2a367